shell/platform/embedder/embedder.h

```diff
@@ -611,6 +611,36 @@ typedef struct {
   int64_t buttons;
 } FlutterPointerEvent;
 
+typedef enum {
+  kFlutterKeyEventKindUp = 1,
+  kFlutterKeyEventKindDown,
+  kFlutterKeyEventKindRepeat,
+} FlutterKeyEventKind;
+
+typedef struct {
+  /// The size of this struct. Must be sizeof(FlutterKeyEvent).
+  size_t struct_size;
+  // Timestamp in microseconds.
```

Comment:
The comment should be very clear about the requirements here. E.g., is using a monotonic clock a requirement, or is it okay for later events to have earlier timestamps when there's a time change?

Reply:
I see the time stamp of key events are defined as `FlutterEngineGetCurrentTime`. Currently I'm using as much time from native as possible in my PRs, wondering if it might provide extra precision or information to applications with certain requirement. Do you recommend me to switch to `FlutterEngineGetCurrentTime`?

Reply:
Not if we don't need it; I just think we should be explicit about what is (or isn't required). If we're okay with native event time, we should say that.

Is it the case that Flutter itself will be fine with non-monotonic event times?